Objects/abstract.c

```diff
@@ -2366,6 +2390,22 @@ PyMapping_GetItemString(PyObject *o, const char *key)
     return r;
 }
 
+int
+PyMapping_GetOptionalItemString(PyObject *o, const char *key, PyObject **pvalue)
+{
+    if (key == NULL) {
+        null_error();
+        return -1;
+    }
+    PyObject *okey = PyUnicode_FromString(key);
+    if (okey == NULL) {
+        return -1;
+    }
+    int r = PyMapping_GetOptionalItem(o, okey, pvalue);
+    Py_DECREF(okey);
+    return r;
```

Comment:
```suggestion
    int res = PyMapping_GetOptionalItem(o, okey, pvalue);
    Py_DECREF(okey);
    return res;
```

Reply:
I have renamed `pvalue` to `result` for consistency with docs and `PyObject_GetOptionalAttr` and `r` to `rc` ("return code").